src/rsa/padding.rs

```diff
@@ -150,3 +150,138 @@ impl Verification for PKCS1 {
         })
     }
 }
+
+/// PSS Padding as described in https://tools.ietf.org/html/rfc3447#section-9.1.
+/// Salt length is fixed to be the length of the output of the digest algorithm.
+/// The only mask generating function supported is MGF1.
+/// The size (in bits) of the RSA moduli is assumed to be a multiple of 8.
+pub struct PSS {
+    pub digest_alg: &'static digest::Algorithm,
+}
+
+// Maxmimum supported output length for PSS padding.
+const MAX_OUTPUT_LEN: usize = 4096;
+
+impl Verification for PSS {
+    // PSS verification as specified in
+    // https://tools.ietf.org/html/rfc3447#section-9.1.2
+    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input)
+              -> Result<(), ()> {
+
+        let em_len = encoded.len();
+        encoded.read_all((), |em| {
+            let digest_len = self.digest_alg.output_len;
+
+            // Step 2.
+            let m_hash = digest::digest(self.digest_alg,
+                                        msg.as_slice_less_safe());
+
+            // Step 3: where we assume the digest and salt are of equal length.
+            if em_len < 2 + (2 * digest_len) {
+                return Err(())
+            }
+
+            // Steps 4 and 5: Parse encoded message as:
+            //     masked_db || h_hash || 0xbc
+            let db_len = em_len - digest_len - 1;
+            let masked_db = try!(em.skip_and_get_input(db_len));
+            let h_hash = try!(em.skip_and_get_input(digest_len));
+            if try!(em.read_byte()) != 0xbc {
+                return Err(());
+            }
+
+            // Step 7.
+            let mut db = [0u8; MAX_OUTPUT_LEN];
```

Comment:
`let db = db[..db_len];`. Then you can use `db.len()` everywhere instead of `db_len`.